addons/web/static/src/js/core/popover.js

```diff
@@ -26,37 +26,77 @@ odoo.define('web.Popover', function () {
             this.state = useState({
                 displayed: false,
             });
+
+            this._onClickDocument = this._onClickDocument.bind(this);
+            this._onScrollDocument = this._onScrollDocument.bind(this);
+            this._onResizeWindow = this._onResizeWindow.bind(this);
+
+            this._onScrollDocument = _.throttle(this._onScrollDocument, 50);
+            this._onResizeWindow = _.debounce(this._onResizeWindow, 250);
+        }
+
+        mounted() {
+            /**
+             * Use capture for the following events to ensure no other part of
+             * the code can stop its propagation from reaching here.
+             */
+            document.addEventListener('click', this._onClickDocument, {
+                capture: true,
+            });
+            document.addEventListener('scroll', this._onScrollDocument, {
+                capture: true,
+            });
+            window.addEventListener('resize', this._onResizeWindow);
```

Comment:
Also, with the previous implementation, we only bound these handlers once, no matter how many popovers there were in the DOM. Why do you need to change that?

Reply:
I think it's not a good idea to couple different components that are otherwise not aware of each other through the use of global variables.
This change allowed to remove the need of syncing a global variable with the state of the currently displayed component, which in turn made the code of the handlers much easier (it was getting even worse if the close/toggle had to be coupled with it).

If we really want to enforce 1 popover open at a time, there should be a root "popover manager" that properly displays the one that should be. And it would probably need to get this information from a store, as "popover targets" can be placed anywhere in the code, typically not as a children of that manager. While this probably should be done for dialogs, I don't think that's needed for popover.

I didn't know about `useExternalListener` but using it will allow to even further simplify the code, so I will try it out.

Reply:
We may have been overzealous, but with list or kanban views, we can quickly reach a larger number of popovers into the DOM, thus leading to a larger number of (global) event handlers. It's easy to imagine a situation where you have 80 records in a list, with a widget having a popover. In this case, every click anywhere in the app, will trigger the execution of 80 handlers. Same for scroll (and less frequently, resize). So I will maybe not lead to performance issues, but it's not elegant at all. And you can still change the limit, so even performance-wise it could get bad.

Also, I agree with you about the PopoverManager. I think it would be excessive.

Actually, we already had this reflection before merging the initial PR, and we considered those 3 solutions. We went for the middle one, which offered the best compromise between performance, and simplicity (having the whole Popover stuff handled by a single Component).

Reply:
Ok, I didn't realize through the comments that it was such an important optimization, and it wasn't covered by tests either.

But now I that understand better, I think I know how to do it to keep that aspect while still avoiding global variables.